lib/auth/machineid/machineidv1/bot_service.go

```diff
@@ -188,6 +190,12 @@ func (bs *BotService) GetBot(ctx context.Context, req *pb.GetBotRequest) (*pb.Bo
 		return nil, trace.Wrap(err, "converting from resources")
 	}
 
+	if err := authCtx.CheckAccessToResource153(
+		bot, types.KindBot, types.VerbRead,
+	); err != nil {
+		return nil, trace.Wrap(err)
```

Comment:
Thoughts on returning "Not Found" rather than "Access Denied" to avoid revealing existence of a resource we do not have permission for?